lib/cache/collections.go

```diff
@@ -41,6 +42,88 @@ type collection interface {
 	watchKind() types.WatchKind
 }
 
+// executor[R] is a specific way to run the collector operations that we need
+// for the genericCollector for a generic resource type R.
+type executor[R types.Resource] interface {
+	getAll(ctx context.Context, cache *Cache, loadSecrets bool) ([]R, error)
+	upsert(ctx context.Context, cache *Cache, value R) error
+	deleteAll(ctx context.Context, cache *Cache) error
+	delete(ctx context.Context, cache *Cache, resource types.Resource) error
+	isSingleton() bool
+}
```

Comment:
I'd like some comments on what these are intended to do, especially for stuff that might be a bit unnatural, like singletons having to return a 1-slice with the value on `getAll`.